src/main/scala/is/hail/table/Table.scala

```diff
@@ -150,18 +208,76 @@ object Table {
 }
 
 class Table(val hc: HailContext, val tir: TableIR) {
+  def this(
+    hc: HailContext,
+    crdd: ContextRDD[RVDContext, RegionValue],
+    signature: TStruct,
+    key: IndexedSeq[String],
+    globalSignature: TStruct,
+    globals: Row
+  ) = this(hc,
+    TableLiteral(
+      TableValue(
+        TableType(signature, key, globalSignature),
+        BroadcastValue(globals, globalSignature, hc.sc),
+        new UnpartitionedRVD(signature, crdd))))
 
-  def this(hc: HailContext,
+  def this(
+    hc: HailContext,
     rdd: RDD[RegionValue],
     signature: TStruct,
-    key: IndexedSeq[String] = Array.empty[String],
-    globalSignature: TStruct = TStruct.empty(),
-    globals: Row = Row.empty) = {
-    this(hc, TableLiteral(
-      TableValue(TableType(signature, key, globalSignature), BroadcastValue(globals, globalSignature, hc.sc),
-        new UnpartitionedRVD(signature, rdd))
-    ))
-  }
+    key: IndexedSeq[String],
+    globalSignature: TStruct,
+    globals: Row
+  ) = this(
+    hc,
+    ContextRDD.weaken(rdd),
+    signature,
+    key,
+    globalSignature,
+    globals)
+
+  def this(
+    hc: HailContext,
+    rdd: RDD[RegionValue],
+    signature: TStruct,
+    key: IndexedSeq[String],
+    globalSignature: TStruct
+  ) = this(hc, rdd, signature, key, globalSignature, Row.empty)
+
+  def this(
+    hc: HailContext,
+    rdd: RDD[RegionValue],
+    signature: TStruct,
+    key: IndexedSeq[String]
+  ) = this(hc, rdd, signature, key, TStruct.empty())
+
+  def this(
+    hc: HailContext,
+    rdd: RDD[RegionValue],
+    signature: TStruct
+  ) = this(hc, rdd, signature, Array.empty[String])
+
+  def this(
+    hc: HailContext,
+    crdd: ContextRDD[RVDContext, RegionValue],
+    signature: TStruct,
+    key: IndexedSeq[String],
+    globalSignature: TStruct
+  ) = this(hc, crdd, signature, key, globalSignature, Row.empty)
+
+  def this(
+    hc: HailContext,
+    crdd: ContextRDD[RVDContext,RegionValue],
+    signature: TStruct,
+    key: IndexedSeq[String]
+  ) = this(hc, crdd, signature, key, TStruct.empty())
+
+  def this(
+    hc: HailContext,
+    crdd: ContextRDD[RVDContext, RegionValue],
+    signature: TStruct
+  ) = this(hc, crdd, signature, Array.empty[String])
```

Comment:
Every possible constructor on either the object or the class?  I have to say I'm against this.  It is just noisy code that a decent convention can eliminate.  The constructor should always take a MatrixType.  I think there should be two constructors, both take that a MatrixType, and one that takes a TableIR and the other (deprecated) that takes a ContextRDD.

This duplicate is solved on Matrix by copyXX routines.  Does that not also work here?